### ISSUE TITLE
Can it be used for text compress?

### ISSUE THREAD

**Issue by blue31606** [WangXuan95/Hard-JPEG-LS#5]
As the title and follow your information, thanks.

**Comment by WangXuan95** [comment]
No, this repo is only for image compression.

For text compression, you can use generic data compression algorithms/formats, e. g., other repo of mine:

(1) An FPGA-based Gzip compressor, with 8300 LUTs and 128MB/s of performance on Xilinx Artix 7:
https://github.com/WangXuan95/FPGA-Gzip-compressor

(2) An FPGA-based LZMA compressor, with 2300 LUTs, 10MB/s of performance, and higher compression ratio on Xilinx Artix 7:
https://github.com/WangXuan95/FPGA-LZMA-compressor

**Comment by blue31606** [comment]
Why is it not suitable for text/data compression? I guess the reason is the algorithm architecture, right?
Because I try to use this repository to compress text/data but the result is bigger than input bytes.

thanks.

**Comment by WangXuan95** [comment]
Yes, because the algorithm is designed for image compression, which use the correlation between pixels (e.g., local smoothing, regular edges) to achieve compression.
However, text data do not have these features, but other features (e.g., limited character set, repeated words). So generic data compression algorithms/formats (e.g. Gzip, LZMA) will work. On the contrary, using image compression algorithm will get even larger data than the original text.
Generic data compression algorithms can also compress image data, but not good as image compression algorithms.

**Comment by blue31606** [comment]
Thanks!
What is the cycle time for LZMA and GZIP to compress 32K bytes? (40MHz CLK)

**Comment by WangXuan95** [comment]
For more information of my GZIP and LZMA FPGA compressor, see README in them.
